### ISSUE TITLE
`comma-style`: does not report trailing commas except in arrays.

### ISSUE THREAD

**Issue by ota-meshi** [eslint-stylistic/eslint-stylistic#598]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Does not report trailing commas except in arrays.

```js
/* eslint "@stylistic/comma-style": ["error", "last", { 
  "exceptions": {
    "ArrayExpression": false,
    "ObjectExpression": false,
  }
}] */
let a, b
const c = [a,b
  ,] // <-- 9:3  error  ',' should be placed last  @stylistic/comma-style
const d = {a,b
  ,} // Not reported.
```

### Reproduction

https://stackblitz.com/edit/github-6uidhe?file=src%2Findex.ts

### Contributes

- [X] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests
